metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -1015,6 +1048,11 @@ private void resetState() {
      */
     private final ControllerPurgatory purgatory;
 
+    /**
+     * A predicate that returns information about whether a ConfigResource exists.
+     */
+    private final ConfigResourceExistenceChecker resourceExists;
```

Comment:
nit: maybe use the less specific type here?

Reply:
ok